### PR TITLE
Relax the regex for addonId to allow any character

### LINKED ISSUE #342: Relax the regex for addonId
This is a spin-off from https://github.com/mozilla/addons-blog/issues/308.

In order to be able to test the blog with the dev API, we need to have a post with add-on identifiers in it that match add-ons on both prod and dev. Currently, prod uses the `id` of the add-on, but we cannot force-create an add-on on dev with a matching `id`, so we would need to use the `slug` or `guid` on dev to match the identifier used in the post. Our code will work if the `slug` or `guid` is used as an identifier in the post, so the easiest way to deal with this is to update a post to use `guid`s instead of `id`s, and then create/update add-ons on dev that have the same `guid`s.

There is currently a regex in [`filters.js`](https://github.com/mozilla/addons-blog/blob/main/src/filters.js#L87) which only allows numeric identifiers right now, so we need to relax that to allow basically any single word. After doing that, a post can be updated to use `guid`s, and then we can set up dev to allow us to test a post with the dev API.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #342

### CODE REVIEW

File: src/filters.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ const makeBetterSafe = ({ markAsSafe }) => {
 const makeBuildStaticAddonCards =
   ({ _buildStaticAddonCard = buildStaticAddonCard } = {}) =>
   async (value, callback) => {
-    const regexp = /<div class="addon-card" data-addon-id="(\d+)"><\/div>/g;
+    const regexp = /<div class="addon-card" data-addon-id="(\S+)"><\/div>/g;
```
Comment (willdurand) [inline src/filters.js:87]: I looked at my notifications in the wrong order so I left a comment in the other PR: https://github.com/mozilla/addons-blog/pull/344#discussion_r759661089

TL;DR: I'd like us to avoid using `\S+` because it is almost like `\s` and that's not very readable. Instead, I'd suggest `.+?` to match everything inside the quotes.

**Review by willdurand — APPROVED** [review]
